internal/lookoutui/src/oidcAuth/OidcAuthProvider.tsx

```diff
@@ -74,11 +74,16 @@ export const OidcAuthProvider = ({ children, oidcConfig }: OidcAuthProviderProps
     }
     const user = await (isOidcRedirectPath ? userManager.signinRedirectCallback() : userManager.getUser())
     if (!user || user.expired) {
-      return await userManager.signinRedirect()
+      return await userManager.signinRedirect({ state: window.location.href })
     }
 
-    setAuthError(undefined)
-    setIsLoading(false)
+    if (isOidcRedirectPath && typeof user.state === "string" && user.state) {
+      const originalURL = new URL(user.state)
+      window.location.replace(`${originalURL.pathname}${originalURL.search}`)
```

Comment:
```suggestion
      // Preserve the current location's host, in case this has been changed by the redirect
      window.location.replace(`${originalURL.pathname}${originalURL.search}`)
```